### ISSUE TITLE
[glew] Build error on arm64-windows

### ISSUE THREAD

**Issue by ArminiusTux** [microsoft/vcpkg#37767]
Package: glew:arm64-windows@2.2.0#3

**Host Environment**

- Host: arm64-windows
- Compiler: MSVC 19.39.33523.0
-    vcpkg-tool version: 2024-02-07-8a83681f921b10d86ae626fd833c253f4f8c355b
    vcpkg-scripts version: a34c873a9 2024-03-22 (5 days ago)

**To Reproduce**

`vcpkg install --x-feature=system-libs --no-print-usage`
**Failure logs**

```
-- Using cached glew-2.2.0.tgz.
-- Cleaning sources at D:/vcpkg/buildtrees/glew/src/glew-80f5997feb.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/vcpkg/downloads/glew-2.2.0.tgz
-- Applying patch fix-LNK2019.patch
-- Applying patch base_address.patch
-- Using source at D:/vcpkg/buildtrees/glew/src/glew-80f5997feb.clean
-- Found external ninja('1.11.0').
-- Configuring arm64-windows-dbg
-- Configuring arm64-windows-rel
-- Building arm64-windows-dbg
-- Building arm64-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Release --target install -- -v -j9
    Working Directory: D:/vcpkg/buildtrees/glew/arm64-windows-rel
    See logs for more information:
      D:\vcpkg\buildtrees\glew\install-arm64-windows-rel-out.log

Call Stack (most recent call first):
  C:/CODING/BUILDS/vcpkg_installed/arm64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  C:/CODING/BUILDS/vcpkg_installed/arm64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  buildtrees/versioning_/versions/glew/cfd894e57985fdb9c772bbf28a4aa261759bb6ac/portfile.cmake:29 (vcpkg_cmake_install)
  scripts/ports.cmake:175 (include)



```
<details><summary>D:\vcpkg\buildtrees\glew\install-arm64-windows-rel-out.log</summary>

```
Change Dir: 'D:/vcpkg/buildtrees/glew/arm64-windows-rel'

Run Build Command(s): "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v -v -j9 install
[1/7] C:/PROGRA~1/CMake/bin/cmcldeps.exe RC D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc CMakeFiles\glew.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res.d CMakeFiles\glew.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res "Note: including file: " "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.39.33519/bin/Hostarm64/arm64/cl.exe" C:\PROGRA~3\WI3CF2~1\10\bin\100226~1.0\arm64\rc.exe -DGLEW_BUILD -DGLEW_NO_GLU -DVC_EXTRALEAN -Dglew_EXPORTS -I D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include -c65001 /DWIN32 /fo CMakeFiles\glew.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc
[2/7] C:/PROGRA~1/CMake/bin/cmcldeps.exe RC D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc CMakeFiles\glew_s.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res.d CMakeFiles\glew_s.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res "Note: including file: " "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.39.33519/bin/Hostarm64/arm64/cl.exe" C:\PROGRA~3\WI3CF2~1\10\bin\100226~1.0\arm64\rc.exe -DGLEW_NO_GLU -DGLEW_STATIC -DVC_EXTRALEAN -I D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include -c65001 /DWIN32 /fo CMakeFiles\glew_s.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc
[3/7] C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\bin\HOSTAR~1\arm64\cl.exe   -DGLEW_BUILD -DGLEW_NO_GLU -DVC_EXTRALEAN -Dglew_EXPORTS -ID:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /MD /O2 /Oi /Gy /DNDEBUG /Z7  -GS- /showIncludes /FoCMakeFiles\glew.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c.obj /FdCMakeFiles\glew.dir\ /FS -c D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c
FAILED: CMakeFiles/glew.dir/D_/vcpkg/buildtrees/glew/src/glew-80f5997feb.clean/src/glew.c.obj 
C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\bin\HOSTAR~1\arm64\cl.exe   -DGLEW_BUILD -DGLEW_NO_GLU -DVC_EXTRALEAN -Dglew_EXPORTS -ID:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /MD /O2 /Oi /Gy /DNDEBUG /Z7  -GS- /showIncludes /FoCMakeFiles\glew.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c.obj /FdCMakeFiles\glew.dir\ /FS -c D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c
D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c(29785) : fatal error C1002: compiler is out of heap space in pass 2
[4/7] C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\bin\HOSTAR~1\arm64\cl.exe   -DGLEW_NO_GLU -DGLEW_STATIC -DVC_EXTRALEAN -ID:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /MD /O2 /Oi /Gy /DNDEBUG /Z7  -GS- /showIncludes /FoCMakeFiles\glew_s.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c.obj /FdCMakeFiles\glew_s.dir\glew_s.pdb /FS -c D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c
FAILED: CMakeFiles/glew_s.dir/D_/vcpkg/buildtrees/glew/src/glew-80f5997feb.clean/src/glew.c.obj 
C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\bin\HOSTAR~1\arm64\cl.exe   -DGLEW_NO_GLU -DGLEW_STATIC -DVC_EXTRALEAN -ID:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /MD /O2 /Oi /Gy /DNDEBUG /Z7  -GS- /showIncludes /FoCMakeFiles\glew_s.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c.obj /FdCMakeFiles\glew_s.dir\glew_s.pdb /FS -c D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c
D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c(29785) : fatal error C1002: compiler is out of heap space in pass 2
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "dependencies": [],
  "features": {
    "system-libs": {
      "description": "System libraries required to build ES.",
      "dependencies": [
        {
          "name": "glew",
          "platform": "!osx"
        },
        "libpng",
        "libjpeg-turbo",
        "libmad",
        {
          "name": "libuuid",
          "platform": "linux"
        },
        "openal-soft",
        {
          "name": "openal-soft",
          "features": [
            "pipewire"
          ],
          "platform": "linux"
        },
        "sdl2"
      ]
    },
    "flatpak-libs": {
      "description": "System libraries missing in the Flatpak runtime.",
      "dependencies": [
        "libmad",
        "glew"
      ]
    },
    "steam-libs": {
      "description": "System libraries not provided in the Steam Sniper Runtime.",
      "dependencies": [
        "libmad"
      ]
    },
    "macos-libs": {
      "description": "System libraries that are always needed on MacOS.",
      "dependencies": [
        "openal-soft"
      ]
    }
  }
}

```
</details>

**Comment by JonLiu1993** [comment]
@ArminiusTux, I can not reproduce your error, if you are using an older version of the C++ compiler, consider upgrading to a newer version that may have better memory management. Please delete any existing build files and directories to ensure a fresh start.

**Comment by ArminiusTux** [comment]
All right, clean start it is - I have removed any sticky traces of "old `glew`" and this is the result:

Package: glew:arm64-windows@2.2.0#3

**Host Environment**

- Host: arm64-windows
- Compiler: MSVC 19.39.33523.0
-    vcpkg-tool version: 2024-03-14-7d353e869753e5609a1f1a057df3db8fd356e49d
    vcpkg-scripts version: 3c76dc55f 2024-03-27 (19 hours ago)

**To Reproduce**

`vcpkg install glew`
**Failure logs**

```
-- Downloading https://github.com/nigels-com/glew/releases/download/glew-2.2.0/glew-2.2.0.tgz -> glew-2.2.0.tgz...
-- Extracting source D:/vcpkg/downloads/glew-2.2.0.tgz
-- Applying patch fix-LNK2019.patch
-- Applying patch base_address.patch
-- Using source at D:/vcpkg/buildtrees/glew/src/glew-80f5997feb.clean
-- Found external ninja('1.11.0').
-- Configuring arm64-windows-dbg
-- Configuring arm64-windows-rel
-- Building arm64-windows-dbg
-- Building arm64-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Release --target install -- -v -j9
    Working Directory: D:/vcpkg/buildtrees/glew/arm64-windows-rel
    See logs for more information:
      D:\vcpkg\buildtrees\glew\install-arm64-windows-rel-out.log

Call Stack (most recent call first):
  installed/arm64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/arm64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/glew/portfile.cmake:29 (vcpkg_cmake_install)
  scripts/ports.cmake:175 (include)



```
<details><summary>D:\vcpkg\buildtrees\glew\install-arm64-windows-rel-out.log</summary>

```
Change Dir: 'D:/vcpkg/buildtrees/glew/arm64-windows-rel'

Run Build Command(s): "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v -v -j9 install
[1/7] C:/PROGRA~1/CMake/bin/cmcldeps.exe RC D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc CMakeFiles\glew.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res.d CMakeFiles\glew.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res "Note: including file: " "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.39.33519/bin/Hostarm64/arm64/cl.exe" C:\PROGRA~3\WI3CF2~1\10\bin\100226~1.0\arm64\rc.exe -DGLEW_BUILD -DGLEW_NO_GLU -DVC_EXTRALEAN -Dglew_EXPORTS -I D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include -c65001 /DWIN32 /fo CMakeFiles\glew.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc
[2/7] C:/PROGRA~1/CMake/bin/cmcldeps.exe RC D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc CMakeFiles\glew_s.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res.d CMakeFiles\glew_s.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res "Note: including file: " "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.39.33519/bin/Hostarm64/arm64/cl.exe" C:\PROGRA~3\WI3CF2~1\10\bin\100226~1.0\arm64\rc.exe -DGLEW_NO_GLU -DGLEW_STATIC -DVC_EXTRALEAN -I D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include -c65001 /DWIN32 /fo CMakeFiles\glew_s.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc.res D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\glew.rc
[3/7] C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\bin\HOSTAR~1\arm64\cl.exe   -DGLEW_BUILD -DGLEW_NO_GLU -DVC_EXTRALEAN -Dglew_EXPORTS -ID:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /MD /O2 /Oi /Gy /DNDEBUG /Z7  -GS- /showIncludes /FoCMakeFiles\glew.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c.obj /FdCMakeFiles\glew.dir\ /FS -c D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c
FAILED: CMakeFiles/glew.dir/D_/vcpkg/buildtrees/glew/src/glew-80f5997feb.clean/src/glew.c.obj 
C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\bin\HOSTAR~1\arm64\cl.exe   -DGLEW_BUILD -DGLEW_NO_GLU -DVC_EXTRALEAN -Dglew_EXPORTS -ID:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /MD /O2 /Oi /Gy /DNDEBUG /Z7  -GS- /showIncludes /FoCMakeFiles\glew.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c.obj /FdCMakeFiles\glew.dir\ /FS -c D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c
D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c(29785) : fatal error C1002: compiler is out of heap space in pass 2
[4/7] C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\bin\HOSTAR~1\arm64\cl.exe   -DGLEW_NO_GLU -DGLEW_STATIC -DVC_EXTRALEAN -ID:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /MD /O2 /Oi /Gy /DNDEBUG /Z7  -GS- /showIncludes /FoCMakeFiles\glew_s.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c.obj /FdCMakeFiles\glew_s.dir\glew_s.pdb /FS -c D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c
FAILED: CMakeFiles/glew_s.dir/D_/vcpkg/buildtrees/glew/src/glew-80f5997feb.clean/src/glew.c.obj 
C:\PROGRA~1\MICROS~1\2022\COMMUN~1\VC\Tools\MSVC\1439~1.335\bin\HOSTAR~1\arm64\cl.exe   -DGLEW_NO_GLU -DGLEW_STATIC -DVC_EXTRALEAN -ID:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\build\cmake\..\..\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /MD /O2 /Oi /Gy /DNDEBUG /Z7  -GS- /showIncludes /FoCMakeFiles\glew_s.dir\D_\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c.obj /FdCMakeFiles\glew_s.dir\glew_s.pdb /FS -c D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c
D:\vcpkg\buildtrees\glew\src\glew-80f5997feb.clean\src\glew.c(29785) : fatal error C1002: compiler is out of heap space in pass 2
ninja: build stopped: subcommand failed.
```
</details>


The step "Configuring arm64-windows-dbg" is done within a few seconds and the next one "Building arm64-windows-rel" hangs for about 5 minutes, while maxing out the cpu at 100% on all 8 cores before eventually failing.

**Comment by JonLiu1993** [comment]
@ArminiusTux, Is this issue still exist in latest vcpkg?

**Comment by ArminiusTux** [comment]
Dear @JonLiu1993 , I am afraid it is - cl.exe takes all the available RAM (up to 7.4 GB) and then fails after 10 minutes:

Package: glew:arm64-windows@2.2.0#3

**Host Environment**

- Host: arm64-windows
- Compiler: MSVC 19.39.33523.0
-    vcpkg-tool version: 2024-04-23-d6945642ee5c3076addd1a42c331bbf4cfc97457
    vcpkg-scripts version: bd2b54836 2024-05-14 (21 hours ago)

**To Reproduce**

`vcpkg install glew`

**Failure logs**

```
-- Using cached glew-2.2.0.tgz.
-- Cleaning sources at D:/vcpkg/buildtrees/glew/src/glew-80f5997feb.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/vcpkg/downloads/glew-2.2.0.tgz
-- Applying patch fix-LNK2019.patch
-- Applying patch base_address.patch
-- Using source at D:/vcpkg/buildtrees/glew/src/glew-80f5997feb.clean
-- Found external ninja('1.12.0').
-- Configuring arm64-windows-dbg
-- Configuring arm64-windows-rel
-- Building arm64-windows-dbg
-- Building arm64-windows-rel
-- Fixing pkgconfig file: D:/vcpkg/packages/glew_arm64-windows/lib/pkgconfig/glew.pc
-- Fixing pkgconfig file: D:/vcpkg/packages/glew_arm64-windows/debug/lib/pkgconfig/glew.pc
-- Installing: D:/vcpkg/packages/glew_arm64-windows/share/glew/vcpkg-cmake-wrapper.cmake
CMake Error at ports/glew/portfile.cmake:73 (file):
  file COPY cannot find "D:/vcpkg/ports/glew/usage": No error.
Call Stack (most recent call first):
  scripts/ports.cmake:175 (include)



```

**Comment by JonLiu1993** [comment]
@ArminiusTux, please delete your previous cache and try again.
